test/parallel_api/iterator/permutation_iterator.h

```diff
@@ -204,6 +204,8 @@ struct test_through_permutation_iterator<TSourceIterator, TSourceDataSize, perm_
     void
     operator()(Operand op)
     {
+
+        auto indexes_begin = dpl::counting_iterator<TSourceDataSize>(0);
```

Comment:
could you please move this line before line 221 ?

Reply:
yes, done. thanks.